java/daikon/dcomp/DCInstrument24.java

```diff
@@ -0,0 +1,4916 @@
+          case LocalVariableType lvt -> {} // we can discard local variable types
+          case LabelTarget l -> {
+            if (ca.labelToBci(l.label()) == 0) {
+              oldStartLabel = l.label();
+            }
+            codeList.add(ce);
+          }
+          default -> codeList.add(ce); // save all other elements
+        }
+      }
+
+      // Create newStartLabel now so instrumentCodeList can use it.
+      newStartLabel = codeBuilder.newLabel();
+
+      // Instrument the method
+      instrumentCodeList(codeModel, mgen, minfo, codeList);
+
+      if (trackMethod && !in_jdk) {
+        add_enter(mgen, minfo, codeList, DCRuntime.methods.size() - 1);
+        add_exit(mgen, mi, minfo, codeList, DCRuntime.methods.size() - 1);
+      }
+
+      // Copy the modified local variable table to the output class.
+      debugInstrument.log("LocalVariableTable:%n");
+      for (LocalVariable lv : mgen.localsTable) {
+        codeBuilder.localVariable(
+            lv.slot(), lv.name().stringValue(), lv.typeSymbol(), lv.startScope(), lv.endScope());
+        @SuppressWarnings("signature:assignment") // need JDK annotations
+        @FieldDescriptor String lvFD = lv.typeSymbol().descriptorString();
+        debugInstrument.log(
+            "  %s : %s%n", lv, daikon.chicory.Instrument24.convertDescriptorToFqBinaryName(lvFD));
+      }
+
+      // Copy the modified instruction list to the output class.
+      ListIterator<CodeElement> li = codeList.listIterator();
+      CodeElement ce;
+      while (li.hasNext()) {
+        if (li.nextIndex() == newStartIndex) {
+          codeBuilder.labelBinding(newStartLabel);
+          debugInstrument.log("Label: %s%n", newStartLabel);
+        }
+        ce = li.next();
+        // If this instruction references a Label, we need to see if it is the oldStartLabel
+        // and, if so, replace the target with the newStartLabel.
+        ce = retargetStartLabel(ce);
+        debugInstrument.log("CodeElement: %s%n", ce);
+        codeBuilder.with(ce);
+      }
+
+      // build_exception_handler returns null if there isn't one.
+      List<CodeElement> handlerCode = build_exception_handler(mgen);
+      if (handlerCode != null) {
+        Label handlerLabel = codeBuilder.newBoundLabel();
+        for (CodeElement ceh : handlerCode) {
+          codeBuilder.with(ceh);
+        }
+        // Using handlerLabel for the end of the try region is technically one instruction
+        // too many, but it shouldn't matter and is easily available.
+        codeBuilder.exceptionCatch(newStartLabel, handlerLabel, handlerLabel, CD_Throwable);
+      }
+    }
+  }
+
+  /**
+   * Returns true if the specified classname.method_name is the root of JUnit startup code.
+   *
+   * @param classname class to be checked
+   * @param method_name method to be checked
+   * @return true if the given method is a JUnit trigger
+   */
+  boolean isJunitTrigger(String classname, String method_name) {
```

Comment:
Inconsistent naming conventions:  use of underscore vs. not.  (Using camelCase rather than an underscore is more idiomatic Java.)